streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -855,6 +856,20 @@ public void advanceWallClockTime(final Duration advance) {
         return new TestOutputTopic<>(this, topicName, keyDeserializer, valueDeserializer);
     }
 
+    /**
+     * Get all the names of all the topics to which records have been produced during the test run.
+     * <p>
+     * Call this method after piping the input into the test driver to retrieve the full set of topic names the topology
+     * produced records to.
+     * <p>
+     * The returned set of topic names includes changelog, repartition and output topic names.
```

Comment:
nit: `changelog, repartition and output topics names` (as there are other internal topics, we might want to rephrase this more general:
```
The returned set of topic names may include user (e.g., output) and internal (e.g., changelog, repartition) topic names.
```